app/controllers/view_components_controller.rb

```diff
@@ -2,7 +2,7 @@
 
 require "rails/application_controller"
 
-class ViewComponentsController < Rails::ApplicationController # :nodoc:
+class ViewComponentsController < ViewComponent::Base.preview_base_controller.constantize # :nodoc:
```

Comment:
I wonder if it would make sense to reverse the inheritance here?

To reverse it, I think we could:

* Always subclass `Rails::ApplicationController`, and custom controllers would be subclasses of `ViewComponentsController`
* Define a `config.view_component.preview_controller` instead of `config.view_component.preview_base_controller`

Then a custom controller could potentially be implemented like this:

```ruby
class MyViewComponentController < ViewComponentsController
  # Define things here
end
```

Then your configuration would look like:

```ruby
config.view_component.preview_base_controller = "MyViewComponentController"
```

I think that change may also make testing this easier.

Thoughts? Would that still solve the problems the existing approach solves?

Reply:
I like this idea! And yes I think it would make testing easier as well.

Reply:
That makes sense to me. I will work on that approach.